src/wasm-lib/kcl/src/std/planes.rs

```diff
@@ -163,10 +166,42 @@ async fn inner_offset_plane(
         }
     }
 
-    Ok(PlaneData::Plane {
-        origin: Box::new(plane.origin),
-        x_axis: Box::new(plane.x_axis),
-        y_axis: Box::new(plane.y_axis),
-        z_axis: Box::new(plane.z_axis),
-    })
+    Ok(plane)
+}
+
+// Engine-side effectful creation of an actual plane object.
+async fn make_offset_plane_in_engine(plane: &Plane, args: &Args) -> Result<uuid::Uuid, KclError> {
+    // Create new default planes.
+    let default_size = 100.0;
+    let color = Color {
+        r: 0.6,
+        g: 0.6,
+        b: 0.6,
+        a: 0.3,
+    };
+
+    args.batch_modeling_cmd(
+        plane.id,
+        ModelingCmd::from(mcmd::MakePlane {
+            clobber: false,
+            origin: plane.origin.into(),
+            size: LengthUnit(default_size),
+            x_axis: plane.x_axis.into(),
+            y_axis: plane.y_axis.into(),
+            hide: Some(false),
+        }),
+    )
+    .await?;
+
+    // Set the color.
+    args.batch_modeling_cmd(
+        uuid::Uuid::new_v4(),
```

Comment:
Need to call the ID generator so that IDs are stable.

```suggestion
        exec_state.id_generator.next_uuid(),
```